docs/publishing-repo.md

```diff
@@ -1,20 +1,58 @@
-    * Consider [obtaining a DOI for your releases](https://guides.github.com/activities/citable-code/), and using that for referencing your code
-
-* If your repository is *private*, ask [tech support](how-to-get-help.md) to make it *public* before the associated paper or output is published. There's more information about what this means [here](repositories.md#repository-access).
-    * Note: you may be prompted to make it public sooner, if you first ran the code against an OpenSAFELY database more than 12 months ago.
+
+
+## Tests
+Try to ensure your automated tests (on the repo's _Actions_ tab) are green.
+It's not essential, but it is a better look to be able to demonstrate your code is minimally runnable.
+
+
+## Contents
+Review your GitHub repository to make it simpler for people to explore:
+
+* Merge or close all open Pull Requests (unless they are ongoing work in progress).
+* Ensure that all branches are merged or deleted apart from the `main` or `master` branch. Old branches may contain draft work or previous ideas that are no longer needed. Note that deleting them does not delete the files permanently, they will still be in the repository history if you need to revisit them later. You may need to leave one or two branches open if they are ongoing work in progress (try to make sure they are named clearly).
+* Delete any files that were not used/experimental and not used in final analysis, e.g. old versions of study definitions. Again they will still be in the repository history if you need them later.
+* Are there old issues that can be closed? Some issues may have been fixed, or may be no longer relevant.
+
+
+## GitHub Release
+* Consider making a GitHub _Release_, and linking to that ([instructions here](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository)).
+A Release is a coherent snapshot of the code as it existed at a given point in time, along with a title and a description.
+By default, releases are displayed on your repo home page.
+For example, you might make separate releases of the code to accompany a preprint and the final versions of the paper.
+
+Consider [obtaining a DOI for your releases](https://guides.github.com/activities/citable-code/), and using that for referencing your code.
+
+
+## Make your repo public
+If your repository is *private*, ask [tech support](how-to-get-help.md) to make it *public* before the associated paper or output is published.
+There's more information about what this means [here](repositories.md#repository-access).
```

Comment:
suggestion: Drop the [use of "here" in links](https://webaim.org/techniques/hypertext/link_text#uninformative). For example:

```suggestion
The [repositories page](repositories.md#repository-access) has more information on controlling repository access.
```